static/js/import_wizard/utils/detection.ts

```diff
@@ -0,0 +1,204 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import _ from "lodash";
+
+import {
+  Column,
+  DCProperty,
+  DCType,
+  MappedThing,
+  Mapping,
+  MappingType,
+  MappingVal,
+  TypeProperty,
+} from "../types";
+
+const COLUMN_KEY = "column";
+const COLUMN_IDX_KEY = "columnIdx";
+const DISP_NAME_KEY = "displayName";
+const PLACETYPE_KEY = "placeType";
+const PLACEPROP_KEY = "placeProperty";
+const DCTYPE_KEY = "dcType";
+const DCPROP_KEY = "dcProperty";
+const HEADER_KEY = "headers";
+
+const COLUMN_EMPTY: Column = { id: "", header: "", columnIdx: 0 };
+const DCPROPERTY_EMPTY: DCProperty = { dcid: "", displayName: "" };
+const DCTYPE_EMPTY: DCType = { dcid: "", displayName: "" };
+
+function _replaceStrings(originalString: string): string {
```

Comment:
functions don't start with underscore in typescript. But I also don't think we need this function

Reply:
I believe it is still needed.